code/CloudFlare.php

```diff
@@ -59,241 +63,102 @@ public function getCFCredentials()
      *
+     * @deprecated This method will be removed in favor for CloudFlare_Purge functionality
+     *
      * @return bool
      */
     public function purgeJavascript()
     {
-        $files = $this->findFilesWithExts(".js");
-
-        if (empty($files)) {
-            $this->setAlert(
-                _t(
-                    "CloudFlare.NoJsFilesFound",
-                    "No Javascript files were found."
-                ),
-                "error"
-            );
-
-            return false;
-        }
-
-        $data = array(
-            "files" => array()
-        );
-
-        foreach ($files as $file) {
-            $data[ 'files' ][] = $this->convertToAbsolute($file);
-        }
+        Deprecation::notice('2.0', 'This method will be removed in favor for CloudFlare_Purge functionality');
 
-        $result   = $this->purgeRequest($data);
-        $response = (is_array($result)) ? $result : json_decode($result, true);
-
-        return $this->responseHandler(
-            $response,
-            _t(
-                "CloudFlare.SuccessPurgeJavascript",
-                "Successfully purged {file_count} JS files from cache.",
-                "",
-                array(
-                    'file_count' => count($data[ 'files' ])
-                )
-            )
-        );
+        return CloudFlareAdmin::singleton()->purge_javascript();
 
     }
 
     /**
      * Finds all image files recursively from root directory, and purges them from cache in a single request
      *
-     * @return bool
+     * @deprecated This method will be removed in favor for CloudFlare_Purge functionality
      */
     public function purgeImages()
     {
-        $files = $this->findFilesWithExts(array(".jpg", ".jpeg", ".gif", ".png", ".ico", ".bmp", ".svg"));
-
-        if (empty($files)) {
-            $this->setAlert(
-                _t(
-                    "CloudFlare.NoImageFilesFound",
-                    "No image files were found."
-                ),
-                "error"
-            );
-
-            return false;
-        }
-
-        $data = array(
-            "files" => array()
-        );
-
-        foreach ($files as $file) {
-            $data[ 'files' ][] = $this->convertToAbsolute($file);
-        }
-
-        $result   = $this->purgeRequest($data);
-        $response = (is_array($result)) ? $result : json_decode($result, true);
-
-        return $this->responseHandler(
-            $response,
-            _t(
-                "CloudFlare.SuccessPurgedImages",
-                "Successfully purged {file_count} image files from cache.",
-                "",
-                array(
-                    'file_count' => count($data[ 'files' ])
-                )
-            )
-        );
+        Deprecation::notice('2.0', 'This method will be removed in favor for CloudFlare_Purge functionality');
 
+        return CloudFlareAdmin::singleton()->purge_images();
```

Comment:
I considered the possibility of branching out from just cache purging into routes etc, ultimately I'd like to keep any purge functionality seperate, I believe that function is also void too so not sure why I bothered returning